webplugin/js/app/media/media-service.js

```diff
@@ -41,5 +41,38 @@ Kommunicate.mediaService = {
 ;            }
         }
     },
+voiceOutputIncomingMessage: function(message){
+    // get appoptions
+    var appOptions = KommunicateUtils.getDataFromKmSession("appOptions");
 
+    // if voiceoutput is enabled and browser supports it
+    if (appOptions.voiceOutput && "speechSynthesis" in window){
+        var textToSpeak = "";
+        if (message) { 
+            if(message.hasOwnProperty("fileMeta")){
+                textToSpeak += MCK_LABELS['voice.output'].attachment;
+                textToSpeak += message.fileMeta.name;
+            }
+            else if (message.contentType == KommunicateConstants.MESSAGE_CONTENT_TYPE.LOCATION){
+                coord = JSON.parse(message.message);
+                textToSpeak +=  MCK_LABELS['voice.output'].location.init;
+                textToSpeak += MCK_LABELS['voice.output'].location.lat + Math.round(coord.lat * 100) / 100;
+                textToSpeak += MCK_LABELS['voice.output'].location.lon + Math.round(coord.lon* 100) / 100;
+
+            }
+            else if (message.message) {
+                textToSpeak += message.message;
+
+            }
+        }
+        if (textToSpeak) {
+        var utterance = new SpeechSynthesisUtterance(textToSpeak);
+        
+        utterance.onerror = ev =>{
+            console.log("Error occured in speech synthesis " + ev.error);
+        };
```

Comment:
@shubhamtibra  We don't print logs in our production builds, so you can throw the error rather than logging.